controler/controler.go

```diff
@@ -119,24 +183,52 @@ func ForwardAuth(c *gin.Context) {
 		Str(realIpHeader, c.Request.Header.Get(realIpHeader)).
 		Msg("Handling forwardAuth request")
 
+	if crowdsecBouncerCacheMode != "none" {
+		entry, err := cache.Get(key)
+		log.Warn().Str("entry", string(entry)).Str("key", string(key)).Msg("Entry")
+
+		if err == nil && len(entry) > 0 {
+			log.Info().Str("Banned", string(string(entry)[0])).Msg("Reading cache")
+			if string(entry)[0] == 'f' {
+				c.Status(http.StatusOK)
+			} else {
+				c.String(crowdsecBanResponseCode, crowdsecBanResponseMsg)
+			}
+			return
+		}
+
+		if crowdsecBouncerCacheMode == "stream" {
+			c.Status(http.StatusOK)
+			return;
+		}
+	}
+
 	// Getting and verifying ip using ClientIP function
-	isAuthorized, err := isIpAuthorized(clientIP)
+	duration, err := getBanDuration(clientIP)
 	if err != nil {
 		log.Warn().Err(err).Msgf("An error occurred while checking IP %q", c.Request.Header.Get(clientIP))
 		c.String(crowdsecBanResponseCode, crowdsecBanResponseMsg)
-	} else if !isAuthorized {
+		return
+	}
+	if duration >= 0 {
+		if crowdsecBouncerCacheMode == "live" && duration != 0 {
+			cache.Set(key, []byte("t"), duration)
+		}
 		c.String(crowdsecBanResponseCode, crowdsecBanResponseMsg)
 	} else {
+		if crowdsecBouncerCacheMode == "live" {
+			cache.Set(key, []byte("f"), int(crowdsecBouncerDefaultCacheDuration.Seconds()))
+		}
 		c.Status(http.StatusOK)
 	}
 }
 
 /*
-	Route to check bouncer connectivity with Crowdsec agent. Mainly use for Kubernetes readiness probe
+Route to check bouncer connectivity with Crowdsec agent. Mainly use for Kubernetes readiness probe
 */
 func Healthz(c *gin.Context) {
```

Comment:
Adding cache to our service break it's statelessness, it should be reflected in this probe.   
Could you implement something like "In stream mode, while the go routine HandleStreamCache hasn't successfully completed a run, this route return a HTTP 503".